WordPress/src/main/java/org/wordpress/android/ui/media/MediaPickerActivity.java

```diff
@@ -453,30 +457,48 @@ public void onAnimationRepeat(Animation animation) {
      */
     public class MediaPickerAdapter extends FragmentPagerAdapter {
         private class MediaPicker {
+            public String loadingText;
+            public String errorText;
+            public String emptyText;
             public String pickerTitle;
             public ArrayList<MediaSource> mediaSources;
 
-            public MediaPicker(String name, ArrayList<MediaSource> sources) {
+            public MediaPicker(String name, String loading, String error, String empty, ArrayList<MediaSource> sources) {
+                loadingText = loading;
+                errorText = error;
+                emptyText = empty;
                 pickerTitle = name;
                 mediaSources = sources;
             }
         }
 
-        private List<MediaPicker> mMediaPickers;
+        private final List<MediaPicker> mMediaPickers;
+        private final List<MediaPickerFragment> mFragments;
 
         private MediaPickerAdapter(FragmentManager fragmentManager) {
             super(fragmentManager);
 
             mMediaPickers = new ArrayList<>();
+            mFragments = new ArrayList<>();
         }
 
         @Override
         public Fragment getItem(int position) {
             if (position < mMediaPickers.size()) {
+                if (position < mFragments.size() && mFragments.get(position) != null) {
```

Comment:
Is this necessary given that `getItem()` is only called when the adapter needs a fragment and the fragment doesn't already exist?